apple/internal/partials/support/resources_support.bzl

```diff
@@ -222,6 +222,37 @@ def _infoplists(ctx, parent_dir, files):
     else:
         return struct(files = [], infoplists = files.to_list())
 
+def _metals(ctx, parent_dir, files):
+    """Processes metal files.
+
+    The metal files will be compiled into a Metal library named `default.metallib`.
+
+    Args:
+        ctx: The target's context.
+        parent_dir: The path under which the library should be placed.
+        files: The metal files to process.
+
+    Returns:
+        A struct containing a `files` field with tuples as described in processor.bzl.
+    """
+    metallib_path = paths.join(parent_dir or "", "default.metallib")
```

Comment:
is the metallib always named `default` ?

Reply:
Xcode does that. I'm not sure about the benefits of allowing the user to specify the name.

Reply:
Looks like it can be customized in Xcode via `METAL_LIBRARY_FILE_BASE` (defined in CoreBuildSystem.xcspec, referenced in `Metal Linker.xcspec`)

Reply:
Allowing to specify an output file name would require a change to the public interface of the rule, which I don’t think we should take easy. I suggest to use the default and hard-coded name for now. What do you think?

Reply:
Maybe make the name a parameter to the metal partial, and have the caller be the one to specify it? Then, presumably, the caller can take ctx into account, and a method can just return default for now, but that way we're set up for it to change later?